server/src-lib/Hasura/Server/Auth/Utils.hs

```diff
@@ -0,0 +1,52 @@
+module Hasura.Server.Auth.Utils
+  ( timeFromCacheControlHeader
+  , timeFromExpiresHeader
+  , toUTCTime
+  , toNominalDiffTime
+  ) where
+
+import           Data.Parser.CacheControl (parseMaxAge)
+import           Data.Time.Clock
+import           Data.Time.Format         (defaultTimeLocale, parseTimeM)
+
+import qualified Data.String.Conversions  as CS
+import qualified Data.Text                as T
+
+import           Hasura.Prelude
+
+
+data ExpiryTime = AbsoluteTime UTCTime
+                | RelativeTime NominalDiffTime
```

Comment:
I don’t think this sum type is useful:

  1. `timeFromCacheControlHeader` has no reason not to return `NominalDiffTime`, since it can never return an `AbsoluteTime`. Returning an `ExpiryTime` makes its type less precise for relatively little gain.

  2. By the same reasoning, `timeFromExpiresHeader` can return a `UTCTime`.

  3. The primary advantage of the sum type seems to be homogenization at call-sites. But both call-sites immediately convert the `ExpiryTime` to either a `UTCTime` or a `NominalDiffTime`, so the values get wrapped only to be immediately unwrapped!

So I think we can drop this sum type and handle each case explicitly at each use site.

Reply:
Done! Yup, it definitely was an attempt at homogenizing call sites, which proved a bad idea in the end.